### PR TITLE
add a new configure item for log_max_files

### PR DESCRIPTION
**Pull request by garyyu**
Currently the log rotation files limitation is hardcoded as 32 log files. That means at maximum we can only keep 512M logs by default (512M here refers to the original file size before gzip), the only way is to change / enlarge `log_max_size` configuration.

This PR add a new config `log_max_files`, default value is still `32`, to replace that hardcoded `count`.

### CODE REVIEW

File: util/src/types.rs [hunk 1]
```diff
@@ -60,6 +62,7 @@ impl Default for LoggingConfig {
 			log_file_path: String::from("grin.log"),
 			log_file_append: true,
 			log_max_size: Some(1024 * 1024 * 16), // 16 megabytes default
+			log_max_files: Some(32),              // 32 log files to rotate over by default
```
Comment (hashmap) [inline util/src/types.rs]: We use `32` in 2 places, would you consider introducing a constant?

**Comment by ignopeverell** [comment]
@garyyu saw the above?

**Comment by garyyu** [comment]
Ah, I saw it but then forgot to change, will modify it now :-)

File: util/src/types.rs (hunk 1 above)
Comment (garyyu) [reply util/src/types.rs]: Done, thanks :-)

**Pull request merged**